assets/js/user_profile.js

```diff
@@ -67,4 +79,24 @@ async function getUserPlans() {
     );
 }
 
-window.onload = getUserPlans;
+async function getUserFavorites() {
+    const favoritesUrl = `/v1/users/${username}/favorites`;
+    await axios.get(
+        favoritesUrl
+    ).then(
+        response => {
+            const data = response.data;
+            const favorites = data['searchHistory'];
+            if (Object.keys(favorites).length > 0) {
+                renderFavorites(favorites);
+            }
+        }
+    ).catch(
+        err => console.error(err)
+    );
+}
+
```

Comment:
wondering what is the difference if we do 
```
async function getUserFavorites() {
	const favoritesUrl = `/v1/users/${username}/favorites`;
	try {
		response = await axios.get(favoritesUrl)
		const data = response.data;
        const favorites = data['searchHistory'];
        if (Object.keys(favorites).length > 0) {
                renderFavorites(favorites);
        }
	}
	catch (error) {
		console.error(`Err: ${err}`)
	}
}
```

Reply:
`Axios` is implicitly doing the `try...catch...`.